crates/router/src/core/payments.rs

```diff
@@ -3364,6 +3379,64 @@ where
     Ok(payment_data)
 }
 
+#[cfg(feature = "v1")]
+pub async fn get_session_token_for_click_to_pay(
+    state: &SessionState,
+    merchant_id: &id_type::MerchantId,
+    key_store: &domain::MerchantKeyStore,
+    authentication_product_ids: serde_json::Value,
+    payment_intent: &hyperswitch_domain_models::payments::PaymentIntent,
+) -> RouterResult<api_models::payments::SessionToken> {
+    use common_utils::id_type::MerchantConnectorAccountId;
+    use hyperswitch_domain_models::payments::ClickToPayMetaData;
+
+    use crate::consts::CLICK_TO_PAY;
+
+    let mca_ids: HashMap<String, MerchantConnectorAccountId> = authentication_product_ids
```

Comment:
we can have these as a type in the database column right? or atleast in the domain models

Reply:
Will do it in a separate PR